src/aclk/mqtt_websockets/mqtt_wss_client.c

```diff
@@ -583,12 +583,18 @@ int mqtt_wss_connect(mqtt_wss_client client, char *host, int port, struct mqtt_c
 
     if (client->sockfd > 0)
         close(client->sockfd);
-    client->sockfd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0);
+    client->sockfd = socket(AF_INET, SOCK_STREAM | DEFAULT_SOCKET_FLAGS, 0);
     if (client->sockfd < 0) {
         mws_error(client->log, "Couldn't create socket()");
         return -1;
     }
 
+#ifdef COMPILED_FOR_MACOS
+    int flags = fcntl(client->sockfd, F_GETFD);
+    flags |= FD_CLOEXEC;
+    (void) fcntl(client->sockfd, F_SETFD, flags);
+#endif
```

Comment:
1. Why we use `COMPILED_FOR_MACOS` and not `#ifndef SOCK_CLOEXEC`?
2. Do we need `if (flags != -1)` check?